pb_plugins/protoc_gen_dcsdk/docs.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+
+
+class Docs(object):
+    """
+    Docs
+
+    Based on https://github.com/protocolbuffers/protobuf/blob/a08b03d4c00a5793b88b494f672513f6ad46a681/src/google/protobuf/descriptor.proto
+    """
+
+    @staticmethod
+    def collect_docs(source_code_info):
+
+        docs = {}
+
+        docs['methods'] = []
+        docs['structs'] = []
+        docs['enums'] = []
+
+        for location in source_code_info.location:
+            if len(location.path) > 0:
+                path = location.path
+
+                if path[0] == 1:  # name
+                    pass
+                elif path[0] == 2:  # package
+                    pass
+                elif path[0] == 4:  # message
+                    struct_id = path[1]
+
+                    if len(path) == 2:  # Top-level location of this message
+                        docs['structs'].append(
+                            {'description': location.leading_comments, 'params': []})
+                    # A field of this message
+                    elif len(path) == 4 and path[2] == 2:
+                        param_id = path[3]
+                        docs['structs'][struct_id]['params'].append(
+                            location.trailing_comments)
+                    elif path[2] == 3:  # A nested message
+                        nested_struct_id = path[3]
+
+                        if len(
+                                path) == 4:  # Top-level location of this nested message
```

Comment:
Weird newline.

Reply:
Agreed, but that's `autopep8` :-/. Good thing is that the style is consistent everywhere, bad thing is that kind of things. I wouldn't know how to improve that, and I think it's not in the scope of this PR.

Reply:
Odd!